### PR TITLE
Feature/sensor history

### PR DESCRIPTION
**Pull request by LisaQ04**
Model and controllers.

### CODE REVIEW

File: WaterLog_Backend/Controllers/SensorHistoryController.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        public SensorHistoryController(DatabaseContext context, IConfiguration config, IControllerService service)
+        {
+            _db = context;
+            _config = config;
+            _service = service;
+        }
+
+        // GET api/sensor history
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<SensorHistoryEntry>>> Get()
```
Comment (FredJEhlers) [inline WaterLog_Backend/Controllers/SensorHistoryController.cs]: More descriptive end point function names please

Comment (FredJEhlers) [reply WaterLog_Backend/Controllers/SensorHistoryController.cs]: Applies to all endpoints

File: WaterLog_Backend/Procedures.cs [hunk 2]
```diff
@@ -1006,6 +1006,13 @@ public enum Period
             arrayOfSeasonsCost.AddPoint("Autum", cost_season[3]);
             return arrayOfSeasonsCost;
         }
+
+        public async Task<double> getTankLevel(int Id)
+        {
+
```
Comment (FredJEhlers) [inline WaterLog_Backend/Procedures.cs]: Is this function supposed to be like this? 
also remove empty lines in the function please

**Review by FredJEhlers — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/SensorHistoryController.cs (hunk 1 above)
Comment (ash-rabbit) [reply WaterLog_Backend/Controllers/SensorHistoryController.cs]: These are auto-generated but yes I agree with Fred. So lets say this one gets sensor history you can rename it to `GetSensorHistory` OR `GetHistory` and the one below this one getting by ID can be renamed to `GetSensoryHistoryById` OR `GetHistoryById`

File: WaterLog_Backend/Controllers/TankLevelsController.cs [hunk 3]
```diff
@@ -38,6 +38,12 @@ public async Task<ActionResult<TankLevelsEntry>> Get(int id)
             return await _db.TankLevels.FindAsync(id);
         }
 
+        [HttpGet("{level}")]
+        public async Task<ActionResult<TankLevelsEntry>> GetLevel(int id)
+        {
+            return await _db.TankLevels.FindAsync(id);
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/TankLevelsController.cs]: Quick question, if DB fails to connect what happens?

**Review by ash-rabbit — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/TankLevelsController.cs (hunk 3 above)
Comment (LisaQ04) [reply WaterLog_Backend/Controllers/TankLevelsController.cs]: Sorry but controller has been changed. No more Tank level.

File: WaterLog_Backend/Procedures.cs (hunk 2 above)
Comment (LisaQ04) [reply WaterLog_Backend/Procedures.cs]: Tank level function no longer exists.

File: WaterLog_Backend/Controllers/SensorHistoryController.cs (hunk 1 above)
Comment (LisaQ04) [reply WaterLog_Backend/Controllers/SensorHistoryController.cs]: Changed function names.

**Review by FredJEhlers — APPROVED** [review]

**Comment by FredJEhlers** [comment]
@LisaQ04 has promised that these requests are being attended to in a teammate's PR 
So @ash-rabbit please update your review

**Review by ash-rabbit — APPROVED** [review]

**Pull request merged**